src/MatchMaker.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  Stack,
+  Box,
+  TextField,
+  Button,
+  List,
+  ListItem,
+  ListItemText,
+} from "@mui/material";
+import { useState } from "react";
+
+type Player = {
+  name: string;
+};
+
+const MatchMaker = () => {
+  const [playerList, setPlayerList] = useState<Player[]>([]);
+  const [playerNameText, setPlayerNameText] = useState<string>("");
+  const addPlayer = () => {
+    const p: Player = { name: playerNameText };
+    setPlayerList([...playerList, p]);
+    setPlayerNameText("");
```

Comment:
```suggestion
    const p: Player = { name: playerNameText };
    if (p.name){
      setPlayerList([...playerList, p]);
      setPlayerNameText("");
    }
```
現状名前入力してなくてもBoxに追加されるから現状わかる対策
本格的な実装ではないはずだから変えなくてもいい
これをもとに作るときは注意って感じ